mmdet/models/utils/panoptic_gt_processing.py

```diff
@@ -51,18 +57,28 @@ def preprocess_panoptic_gt(gt_labels, gt_masks, gt_semantic_seg, num_things,
     for label in semantic_labels:
         if label < num_things or label >= num_classes:
             continue
-        stuff_mask = gt_semantic_seg == label
+        stuff_mask = (gt_semantic_seg == label).long()
         stuff_masks_list.append(stuff_mask)
         stuff_labels_list.append(label)
 
-    if len(stuff_masks_list) > 0:
-        stuff_masks = torch.stack(stuff_masks_list, dim=0)
-        stuff_labels = torch.stack(stuff_labels_list, dim=0)
-        labels = torch.cat([things_labels, stuff_labels], dim=0)
-        masks = torch.cat([things_masks, stuff_masks], dim=0)
+    if return_seperate:
+        if len(stuff_labels_list) > 0:
+            stuff_masks = torch.stack(stuff_masks_list, dim=0)
+            stuff_labels = torch.stack(stuff_labels_list, dim=0)
+        else:
+            stuff_masks = gt_semantic_seg.new_zeros(
+                size=(0, ) + gt_semantic_seg.shape[-2:])
+            stuff_labels = gt_semantic_seg.new_zeros(size=(0, ))
+
+        return things_labels, things_masks, stuff_labels, stuff_masks
```

Comment:
the return is different, may occur some potential error. Maybe we can add `None` to force set return value that not needed when return_seperate is False